src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -94,6 +98,9 @@ protected Workspace(HostServices host, string? workspaceKind)
             _latestSolution = CreateSolution(info, emptyOptions, analyzerReferences: SpecializedCollections.EmptyReadOnlyList<AnalyzerReference>());
 
             _optionService.RegisterDocumentOptionsProvider(EditorConfigDocumentOptionsProviderFactory.Create());
+
+            var exportProvider = (IMefHostExportProvider)this.Services.HostServices;
```

Comment:
This would break any third party hosts that are non-MEF based...I'm not sure how many of those actually exist in the real world (could very likely be zero) but the cast here is architecture violation here.

Reply:
I couldn't see any other way to get a normal MEF service here, and adding something to the constructor seemed a little aggressive, so I searched and found this. Is there a better way? Or do you just mean this shouldn't assume the cast will succeed?